File: python/robot-name/robot_name.py
from random import choice, randint
import string
class Robot:
    names = set()
    def __init__(self):
        self.reset()

    def rand_name(self):
        letters = string.ascii_uppercase
        name = choice(letters) + choice(letters)
        for i in range(3):
            name += str(randint(0,9))
        if name in Robot.names: return self.rand_name()
        else: Robot.names.add(name)
        return name

    def reset(self):
        self.name = self.rand_name()
        return self.name

File: swift/robot-name/Sources/RobotName/RobotName.swift
class Robot {
    let letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    let ints = "1234567890"
    var name:String = ""

    func resetName() {
        name = ""
        for _ in 1...2 { name += String(letters.randomElement()!) }
        for _ in 1...3 { name += String(ints.randomElement()!) }
    }

    init(){
        resetName()
    }
}
